### ISSUE TITLE
“Detected 0 devices” 

### ISSUE THREAD

**Issue by SmallLilDemon** [nanopool/nanominer#152]
Hello. I just set up a mining machine with 6 5700 xt’s. Using windows 10. Just updated everything and verified that the gpus are being detected (They are on 3 different programs). I set up the wallet in the config_eth as im going to be mining eth. When i launch the nanominer application, i get “detected 0 devices”. “Press enter to continue.” When i press enter, the application closes. Any ideas as to why the program isnt detecting the gpus?

**Comment by PlancherFlottant** [comment]
I have the same problem with my AMD Radeon RX 5600 XT. I just set up everything right. I tryed everything I could find and nothing helps! 

Im at the same place than SmallLilDemon

**Comment by PlancherFlottant** [comment]
@Grumpy-Dwarf  we need your help!

**Comment by PlancherFlottant** [comment]
@nanopool  @nanominer

**Comment by Grumpy-Dwarf** [comment]
What OS and AMD driver version are you using?

**Comment by darkxion** [comment]
> Hello. I just set up a mining machine with 6 5700 xt’s. Using windows 10. Just updated everything and verified that the gpus are being detected (They are on 3 different programs). I set up the wallet in the config_eth as im going to be mining eth. When i launch the nanominer application, i get “detected 0 devices”. “Press enter to continue.” When i press enter, the application closes. Any ideas as to why the program isnt detecting the gpus?

I am having the exact same problem. 1 windows 10 machine with a single RX 580

**Comment by PlancherFlottant** [comment]
I have communication with a nanomimer tech! I Jeep you in touch

**Comment by SmallLilDemon** [comment]
> What OS and AMD driver version are you using?

I am using windows 10 pro and for the drivers what ever the newest drives are for the 5700xt i will look at exactly what version they are when i get home.

**Comment by PlancherFlottant** [comment]
> What OS and AMD driver version are you using?

I installed yesterday the AMS Radeon software ans Everthing is Up to date.

5600 xt BIOS version: 017.001.000.068
Driver version: See picture

![MVIMG_20210207_101804](https://user-images.githubusercontent.com/78657967/107234763-fe55c200-69f1-11eb-91f1-2fa0029bf2b2.jpg)

**Comment by PlancherFlottant** [comment]
All this on windows 10 home 
version 10.0.18363 build 18363

**Comment by SmallLilDemon** [comment]
> > What OS and AMD driver version are you using?
> 
> I installed yesterday the AMS Radeon software ans Everthing is Up to date.
> 
> 5600 xt BIOS version: 017.001.000.068
> Driver version: See picture
> 
> ![MVIMG_20210207_101804](https://user-images.githubusercontent.com/78657967/107234763-fe55c200-69f1-11eb-91f1-2fa0029bf2b2.jpg)

I as well just installed amd’s program for the card. Going to be home today so i will take a look at my driver version as well :)

**Comment by SmallLilDemon** [comment]
> What OS and AMD driver version are you using?

Driver for my card is: Adrenalin 2020 Edition 21.1.1

**Comment by Grumpy-Dwarf** [comment]
Still unable to reproduce, however if remove AMD driver using DDU and fresh install 21.2.1 does miner see devices?

**Comment by Grumpy-Dwarf** [comment]
Anyway, I've updated AMD GPUs detection code in new 3.2.1 release, hope this fixes the problem. If not, please notify. If someone has found a workaround (reinstalling with DDU probably), please share it.
https://github.com/nanopool/nanominer/releases/tag/v3.2.1

**Comment by PlancherFlottant** [comment]
Works well! Thanx for that and your speed of response :D Great job

**Comment by PlancherFlottant** [comment]
Seems like avast antivirus make the error to come back! When avst activate it shows 0 devices detected on the nanominer!

**Comment by SmallLilDemon** [comment]
> Seems like avast antivirus make the error to come back! When avst activate it shows 0 devices detected on the nanominer!



> Anyway, I've updated AMD GPUs detection code in new 3.2.1 release, hope this fixes the problem. If not, please notify. If someone has found a workaround (reinstalling with DDU probably), please share it.
> https://github.com/nanopool/nanominer/releases/tag/v3.2.1

Thank you very much. I will try it out!

**Comment by SmallLilDemon** [comment]
> Seems like avast antivirus make the error to come back! When avst activate it shows 0 devices detected on the nanominer!

Really? Well thats good to know. Guess ill have to delete my avast and install another program. I could have sworn though that there was a way to allow programs to pass through avast.

**Comment by Grumpy-Dwarf** [comment]
Do all people with this issue have Avast installed?

**Comment by smetigo** [comment]
Hello, I am facing the same issue. I do have Avast installed. Did you try to run mining with another software? I did with lolMiner and I have 8 MH/s only with RX 570. Should be around 24

**Comment by RunPi314** [comment]
Turning Avast off while I started the nanominer solved the issue for me.

**Comment by SmallLilDemon** [comment]
Yes. Avast was the issue. Deleted it and nashminer works.

**Comment by GaryFAlderson** [comment]
I am having the same problem since upgrading to Ubuntu 22.04. I tried three different amdgpu-installs. 
amdgpu-install_22.20.50200-1_all.deb  amdgpu-install_5.3.50300-1_all.deb
amdgpu-install_22.20.50204-1_all.deb
The only one that actually worked was amdgpu-install_5.3.50300-1_all.deb which is the "non-pro" open source version.
All sanity checks passed but I do not have vulkan installed (yet)
Install: apt-get install amdgpu-install_5.3.50300-1_all.deb
amdgpu-install --usecase=graphics --accept-eula

I have tried nanominer 3.7.3 (which worked fine under Ubuntu 20.04.5) and also nanominer 3.7.5
Both nanominer -d gave me Dtected 0 devices
I have 5 MSI/AMD RX 470's which also worked fine beforehand.

Has there been any progress on this ticket ? Do I need to install the vulkan sdk ?
